src/mds/PurgeQueue.cc

```diff
@@ -43,7 +56,12 @@ void PurgeItem::encode(bufferlist &bl) const
 
 void PurgeItem::decode(bufferlist::const_iterator &p)
 {
-  DECODE_START(1, p);
+  DECODE_START(2, p);
+  if (struct_v >= 2) {
+    decode(stamp, p);
+    decode(pad_size, p);
+    p.advance(pad_size);
+  }
```

Comment:
encoding action/ino/size after padding. does this work for JournalTool::erase_region